libpod/container_internal_linux.go

```diff
@@ -1458,11 +1458,35 @@ func (c *Container) generateCurrentUserPasswdEntry() (string, error) {
 	if uid == 0 {
 		return "", nil
 	}
+
 	u, err := user.LookupId(strconv.Itoa(rootless.GetRootlessUID()))
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to get current user")
 	}
-	return fmt.Sprintf("%s:x:%s:%s:%s:%s:/bin/sh\n", u.Username, u.Uid, u.Gid, u.Username, c.WorkingDir()), nil
+
+	// Lookup the user to see if it exists in the container image.
+	_, err = lookup.GetUser(c.state.Mountpoint, u.Username)
```

Comment:
What if the user does exist inside the container, but with a different UID?

Reply:
I think we end up with duplicate entries in that case, with this one being the second of the two. I don't think that's *completely* disastrous, but it's definitely not ideal.